pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -1008,15 +1010,26 @@ func k3sClusterConfig(cluster *v3.Cluster, nodes []*v3.Node) error {
 		}
 		// only set these values on init
 		if cluster.Spec.K3sConfig == nil {
-			cluster.Spec.K3sConfig = &v32.K3sConfig{
-				Version: cluster.Status.Version.String(),
-				K3sUpgradeStrategy: v32.K3sUpgradeStrategy{
-					ServerConcurrency: 1,
-					WorkerConcurrency: 1,
-				},
-			}
+			cluster.Spec.K3sConfig = &v32.K3sConfig{}
+			setTypeConfig(cluster.Spec.K3sConfig, cluster.Status.Version.String())
+		}
+	} else if strings.Contains(cluster.Status.Version.String(), "rke2") {
+		cluster.Status.Driver = v32.ClusterDriverRke2
+		if cluster.Spec.Rke2Config == nil {
+			cluster.Spec.Rke2Config = &v32.Rke2Config{}
+			setTypeConfig(cluster.Spec.Rke2Config, cluster.Status.Version.String())
 		}
 	}
-
 	return nil
 }
+
+func setTypeConfig(config interface{}, version string) {
+	switch c := config.(type) {
+	case *v32.K3sConfig:
+		c.Version = version
+		c.SetStrategy(1, 1)
+	case *v32.Rke2Config:
+		c.Version = version
+		c.SetStrategy(1, 1)
+	}
+}
```

Comment:
Couldn't you just do this above: 
```
cluster.Spec.K3sConfig = &v32.K3sConfig{
  Version: cluster.Status.Version.String(),
}
cluster.Spec.K3sConfig.SetStrategy(1,1)
```

Seems simpler. Passing a pointer to an interface and then switching feels weird. Actually I liked the old way best, without the `SetStrategy` setter method where you just hardcoded everything inline. Less boilerplate code and passing pointers around.